Lib/idlelib/idle_test/test_editor.py

```diff
@@ -189,6 +193,42 @@ def test_indent_and_newline_event(self):
         nl(None)
         eq(get('1.0', 'end'), '>>> \na =\n')
 
+    def test_parser(self):
```

Comment:
Let's split into test_parser_editor and test_parser_shell.  Even after prompt is removed from editing area, we may want separate handling.

At the moment, 
`>>> import sys; sys.ps1=''; import idlelib.idle`
runs IDLE with prompt_last_line == '', so the editor code is used in shell.  But it still seems to work, and prompt will be fixed to '>>>' or at least something non-empty when moved to sidebar.